docs/proposals/diagnose-experience.md

```diff
@@ -0,0 +1,247 @@
+
+Given the diverse roles and scenarios in which ORAS CLI is utilized, we have received feedback from users and developers to improve the diagnostic experience. Enhancing the verbose output and debug logs can significantly benefit ORAS users and developers by making diagnostics clearer and more unambiguous.
+
+This proposal document aims to:
+
+1. Identify the issues associated with the current implementation of the `--verbose` and `--debug` options.
+2. Clarify the concepts of verbose output and debug logs.
+3. List the guiding principles to write comprehensive, clear, and conducive verbose output and debug logs for effective diagnosis.
+4. Propose solutions to improve the diagnostic experience for ORAS CLI users and developers.
+
+## Problem Statement
+
+Specifically, there are exiting GitHub issues raised in the ORAS community.
+
+- The user is confused about when to use `--verbose` and `--debug`. See the relevant issue [#1382](https://github.com/oras-project/oras/issues/1382).
+- Poor readability of debug logs. No separator lines between request and response information. Users need to add separator lines manually for readability. See the relevant issue [#1382](https://github.com/oras-project/oras/issues/1382).
+- Critical information is missing in debug logs. For example, the [error code](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#error-codes) and metadata of the processed resource object (e.g. image manifest) are not displayed.
+- The detailed operation information is missing in verbose output. For example, how many resource objects are processed where. Less or none verbose output of ORAS commands in some use cases.
+- Timestamp of each request and response is missing in debug logs, which is hard to trace historical operation and trace the sequence of events accurately.
+
+## Concepts
+
+Before restructuring the log information of ORAS debug log and [verbose output](https://en.wikipedia.org/wiki/Verbose_mode), it worth clarifying the concepts and differences between verbose output and debug logs.
+
+### Verbose Output 
+
+Verbose output focuses on providing a comprehensive, high-level view of the application's operations. It is intended for end-users who want to observe the detailed normal operation of the tool. It should be human readable and descriptive.
+
+- **Purpose**: Verbose output are typically used to provide highly detailed information about the flow and operations of the application. They give users a comprehensive view of what the tool is doing at every step.
+- **Target users**: These are generally intended for users who want to understand the detailed workings of the tool, not necessarily for debugging specific issues.
```

Comment:
+1. A user trying to copy an artifact might experience failure and is trying to debug their failure. Hence the tool should provide verbose logs to help debug scenarios and gain insight into the overall flow of operations.

Reply:
Per our community meeting discussion, ORAS maintainers agreed to deprecate `--verbose` and remain `--debug` as a unified control of debug logs. I updated the spec to avoid ambiguity.